### PR TITLE
Arithmetic intensity integration

### PR DESCRIPTION
**Pull request by fkeceli**
Integration of the Arithmetic Intensity benchmark. 

Downloaded from https://github.com/dannosliwcd/arithmetic-intensity

Uses the nasm utility: https://github.com/netwide-assembler/nasm

### CODE REVIEW

File: integration/apps/arithmetic_intensity/build.sh [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+set -x
+set -e
+
+# Get helper functions
+source ../build_func.sh
+
+if [ ! -x "$(command -v nasm)" ]; then
+    DIRNAME=nasm
+    GITREPO=https://github.com/netwide-assembler/nasm.git
```
Comment (dannosliwcd) [inline integration/apps/arithmetic_intensity/build.sh]: This seems like a pretty active repo, which means the top hash could change pretty frequently. Might want to wget one of the releases instead: https://github.com/netwide-assembler/nasm/releases

File: integration/apps/arithmetic_intensity/build.sh [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    unpack_archive ${ARCHIVE}
+else
+    clone_repo_git ${GITREPO} ${DIRNAME} ${TOPHASH}
+fi
+setup_source_git ${DIRNAME}
+
+# build
+echo $PATH
+cd ${DIRNAME}
+make
```
Comment (dannosliwcd) [inline integration/apps/arithmetic_intensity/build.sh]: Can `make CXX=$MPICXX`, then no patch is needed.

File: integration/experiment/monitor/run_monitor_arithmetic_intensity.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+#  SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+#  LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+#  DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+#  THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+#  (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY LOG OF THE USE
+#  OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+
+'''
+Run Pennant with the monitor agent.
```
Comment (dannosliwcd) [inline integration/experiment/monitor/run_monitor_arithmetic_intensity.py]: Pennant -> arithmetic intensity

File: integration/apps/arithmetic_intensity/build.sh (hunk 1 above)
Comment (fkeceli) [reply integration/apps/arithmetic_intensity/build.sh]: Done.

File: integration/apps/arithmetic_intensity/build.sh (hunk 2 above)
Comment (fkeceli) [reply integration/apps/arithmetic_intensity/build.sh]: Done.

File: integration/experiment/monitor/run_monitor_arithmetic_intensity.py (hunk 3 above)
Comment (fkeceli) [reply integration/experiment/monitor/run_monitor_arithmetic_intensity.py]: Fixed.

**Review by dannosliwcd — APPROVED** [review]

**Pull request merged**